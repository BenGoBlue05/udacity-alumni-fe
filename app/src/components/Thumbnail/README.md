## Thumbnail Component
A component that ...

### Example

```js
<Thumbnail />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
