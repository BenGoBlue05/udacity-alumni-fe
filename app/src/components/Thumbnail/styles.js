import styled from 'styled-components';
import Image from 'grommet-udacity/components/Image';
import Box from 'grommet-udacity/components/Box';

const sizeMap = (size) => {
  switch(size) {
    case 'xsmall':
      return 50;
    case 'small':
      return 100;
    case 'medium':
      return 150;
    case 'large':
      return 200;
    case 'xlarge':
      return 300;
    default:
      return 100;
  }
};

export const Wrapper = styled(Box)`
  width: ${props => sizeMap(props.imageSize)}px;
`;

export const ThumbnailImage = styled(Image)`
  border: solid 4px #FFFFFF;
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.2);
  display: inline-block;
  line-height: 0;
  max-width: 100%;
  height: auto;
  transition: all 200ms ease-out;
`;
