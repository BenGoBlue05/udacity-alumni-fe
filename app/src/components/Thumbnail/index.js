import React, { PropTypes } from 'react';
import { ThumbnailImage, Wrapper } from './styles';
const defaultAvatarUrl = 'https://github.com/RyanCCollins/cdn/blob/master/misc/placeholder3.png?raw=true';

const Thumbnail = ({
  src,
  size,
}) => (
  <Wrapper imageSize={size}>
    <ThumbnailImage
      src={src || defaultAvatarUrl}
    />
  </Wrapper>
);

Thumbnail.propTypes = {
  src: PropTypes.string.isRequired,
  size: PropTypes.oneOf([
    'xsmall',
    'small',
    'medium',
    'large',
    'xlarge',
  ]),
};

Thumbnail.defaultProps = {
  src: defaultAvatarUrl,
  size: 'small',
};

export default Thumbnail;
