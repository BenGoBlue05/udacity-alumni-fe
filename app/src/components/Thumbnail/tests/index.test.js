import Thumbnail from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<Thumbnail />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Thumbnail />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
