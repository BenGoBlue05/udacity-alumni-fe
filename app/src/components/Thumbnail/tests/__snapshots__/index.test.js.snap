exports[`<Thumbnail /> should render with default props 1`] = `
<Styled(Box)
  imageSize="small">
  <Styled(Image)
    src="https://github.com/RyanCCollins/cdn/blob/master/misc/placeholder3.png?raw=true" />
</Styled(Box)>
`;
